shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala

```diff
@@ -709,19 +709,19 @@ object GlutenConfig {
       .internal()
       .doc(
         "By default, the supported codecs are lz4 and zstd. " +
-          "When spark.gluten.sql.columnar.shuffle.codecBackend=qat, the supported codec is gzip. " +
-          "When spark.gluten.sql.columnar.shuffle.codecBackend=iaa, the supported codec is gzip.")
+          "When spark.gluten.sql.columnar.shuffle.codecBackend=qat," +
+          "the supported codecs are gzip and zstd. " +
+          "When spark.gluten.sql.columnar.shuffle.codecBackend=iaa," +
+          "the supported codec is gzip.")
       .stringConf
-      .transform(_.toUpperCase(Locale.ROOT))
-      .checkValues(
-        GLUTEN_SHUFFLE_SUPPORTED_CODEC ++ GLUTEN_QAT_SUPPORTED_CODEC ++ GLUTEN_IAA_SUPPORTED_CODEC)
+      .transform(_.toLowerCase(Locale.ROOT))
```

Comment:
also here

Reply:
Here we want it to be converted when user provides uppercase config.

Reply:
Same as this one, https://github.com/oap-project/gluten/pull/2197#discussion_r1257735246 If we only accept lower case, we don't need to transform.